### PR TITLE
hydroclimav2: name the STS status word bits and TPL event groups

### PR DESCRIPTION
**Pull request by strnadmichal**
Answers the `Help needed: what do the STS bits mean?` comment in the driver.

**Sources.** No BMeters protocol spec for the STS bits was available to us. We worked from:

1. *Hydroclima 2 User Manual v1.1*, section "Alarms" ([PDF](https://www.tzbtechnika.cz/user/related_files/pomerovy-indikator-uzivatelsky-navod-maddeo-bmeters-hydroclima-2-rfm.pdf), p. 8) — documents an error/event report "in the status field" with events labelled **A0, A1, B8, C9, C10, C11, C13, C15**. The numbers 0–15 fit a 16-bit word exactly, so we read them as bit positions.
2. Empirical correlation on 9 decrypted telegrams from a production fleet (BMP, fw 0x85): TPL `0x40` ↔ word `0x0100` (3×), TPL `0x80` ↔ word `0x0200` (2×), TPL `0xC0` ↔ word `0x0300` (2×), TPL `OK` ↔ word `0x0000` (2×). No counterexamples.

**Confidence.** Word-bit names: high (manual-documented events; positions inferred from their numbering). TPL `EVENT_GROUP_B/C`: we only ever observed events B8 and C9, so we cannot distinguish "group flag" from "specific event flag" — we chose group naming because the manual groups events by letter; happy to rename if anyone has the BMeters spec. `EVENT_GROUP_A` (0x20) is extrapolated and never observed. Bits 2–7, 12 and 14 are undocumented and stay unnamed (one shows up in the existing 0x53 test telegram as `ERROR_FLAGS_4000`).

Driver tests updated accordingly; `make pretty` applied. Corrections from anyone holding BMeters documentation are very welcome.

### CODE REVIEW

**Comment by weetmuts** [comment]
Nice, is this PR still a work in progress?

**Review by Hymbajs — COMMENTED** [review]
Great work — I checked this against the primary source you cited (Hydroclima 2 User Manual v1.1, "Alarms" table, p.8) and every one of the 8 STS bit positions and values matches exactly (A0=0x0001 ... C15=0x8000), including the TPL event-group extrapolation. I also found independent corroboration in the existing test telegrams: in `93001021`, STS=0x0100 pairs with a non-zero `DOP` (housing-open date) field, and in `93000952`, STS=0x0000 pairs with `DOP`=0x0000 — consistent with bit 8 being the case-open event.

I compared the three new names against similar flags in the rest of the driver set for consistency and left inline suggestions for three renames (all mechanical, values unchanged) — details on each comment. Verified locally: `make pretty` clean, `cd drivers && make testonly DRIVER=hydroclimav2` passes with the suggested test updates.

File: drivers/src/hydroclimav2.xmq [hunk 1]
```diff
@@ -27,9 +54,53 @@ driver {
                 map_type        = BitToString
                 mask_bits       = 0xffff
                 default_message = OK
-                // The STS status word bits are not documented by BMeters.
-                // Surface the raw value so faults are visible; name the bits later.
-                // Help needed: what do the STS bits mean?
+                // STS status word bits per the Hydroclima 2 user manual v1.1
+                // (Alarms table): events A0, A1, B8, C9, C10, C11, C13 and
+                // C15 are bit positions in this word. Bits 2-7, 12 and 14 are
+                // not documented and stay unnamed. OPEN_CASE latches until a
```
Comment (Hymbajs) [inline drivers/src/hydroclimav2.xmq]: Nit: update this comment to match the renamed bit below.

```suggestion
                // not documented and stay unnamed. TAMPERING latches until a
```

File: drivers/src/hydroclimav2.xmq [hunk 2]
```diff
@@ -27,9 +54,53 @@ driver {
...
+                    value = 0x0001
+                    test  = Set
+                }
+                map {
+                    name  = VERIFICATION_PENDING
+                    value = 0x0002
+                    test  = Set
+                }
+                map {
+                    name  = OPEN_CASE
```
Comment (Hymbajs) [inline drivers/src/hydroclimav2.xmq]: **Naming consistency with the rest of the codebase.** `TAMPER`/`TAMPERING` is the established name for a case/cover-open event across this driver set (`kaden.xmq`, `kamwater.xmq`, `ultrimis.xmq`/`ultrimisv2.xmq`, `minomess.xmq`, `iwmtx5.xmq`, `ei6500.xmq`, `aquastream.xmq`). `kaden.xmq` is the closest analogue here — it's also a heat cost allocator, and it also pairs the flag with a recorded date/duration (`tampering_duration_h`), just like this driver's `DOP` field. "CASE" doesn't otherwise appear as a term anywhere in this repo's driver names.

```suggestion
                    name  = TAMPERING
```

File: drivers/src/hydroclimav2.xmq [hunk 3]
```diff
@@ -27,9 +54,53 @@ driver {
...
+                    value = 0x0100
+                    test  = Set
+                }
+                map {
+                    name  = TEMPERATURE_OUT_OF_RANGE
```
Comment (Hymbajs) [inline drivers/src/hydroclimav2.xmq]: **Disambiguate from `average_ambient_temperature`.** The manual's C9 event is specifically about the radiator/TK sensor ("Permitted temperature range is <5°C...95°C>" for the *radiator* temperature parameter), not the ambient sensor. This driver already names that sensor "heater" everywhere else (`average_heater_temperature`, `average_heater_temperature_last_month`, `AverageHeater` in the ixml grammar) — "radiator" doesn't appear anywhere in this file. Naming the flag after the heater sensor keeps it internally consistent and avoids readers assuming it refers to `average_ambient_temperature_c`.

```suggestion
                    name  = HEATER_TEMPERATURE_OUT_OF_RANGE
```

File: drivers/src/hydroclimav2.xmq [hunk 4]
```diff
@@ -27,9 +54,53 @@ driver {
...
+                    value = 0x0200
+                    test  = Set
+                }
+                map {
+                    name  = RADIATOR_TEMPERATURE_LOW
```
Comment (Hymbajs) [inline drivers/src/hydroclimav2.xmq]: Same point as the C9 flag above — "radiator" isn't used anywhere else in this driver; this driver's own term for that sensor is "heater".

```suggestion
                    name  = HEATER_TEMPERATURE_LOW
```

File: drivers/src/hydroclimav2.xmq [hunk 5]
```diff
@@ -239,8 +310,8 @@ driver {
...
             telegram = 2e44b0099861036853087a000020002F2F036E0000000F100043106A7D2C4A078F12202CB1242A06D3062100210000
-            json     = '{"_":"telegram","media":"heat cost allocation","meter":"hydroclimav2","name":"HCA","id":"68036198","average_ambient_temperature_c":18.66,"average_ambient_temperature_last_month_c":15.78,"average_heater_temperature_last_month_c":17.47,"consumption_hca":0,"max_ambient_temperature_c":47.51,"total_consumption_hca":33,"status":"ERROR_FLAGS_4300","timestamp":"1111-11-11T11:11:11Z"}'
-            fields   = 'HCA;68036198;ERROR_FLAGS_4300;0;18.66;1111-11-11 11:11.11'
+            json     = '{"_":"telegram","media":"heat cost allocation","meter":"hydroclimav2","name":"HCA","id":"68036198","average_ambient_temperature_c":18.66,"average_ambient_temperature_last_month_c":15.78,"average_heater_temperature_last_month_c":17.47,"consumption_hca":0,"max_ambient_temperature_c":47.51,"total_consumption_hca":33,"status":"ERROR_FLAGS_4000 OPEN_CASE TEMPERATURE_OUT_OF_RANGE","timestamp":"1111-11-11T11:11:11Z"}'
+            fields   = 'HCA;68036198;ERROR_FLAGS_4000 OPEN_CASE TEMPERATURE_OUT_OF_RANGE;0;18.66;1111-11-11 11:11.11'
```
Comment (Hymbajs) [inline drivers/src/hydroclimav2.xmq]: Update the test expectation for the renames above. Note `BitToString` output is alphabetically sorted, not declaration-order, so `HEATER_TEMPERATURE_OUT_OF_RANGE` sorts before `TAMPERING` here (verified locally with `make testonly DRIVER=hydroclimav2`).

```suggestion
            json     = '{"_":"telegram","media":"heat cost allocation","meter":"hydroclimav2","name":"HCA","id":"68036198","average_ambient_temperature_c":18.66,"average_ambient_temperature_last_month_c":15.78,"average_heater_temperature_last_month_c":17.47,"consumption_hca":0,"max_ambient_temperature_c":47.51,"total_consumption_hca":33,"status":"ERROR_FLAGS_4000 HEATER_TEMPERATURE_OUT_OF_RANGE TAMPERING","timestamp":"1111-11-11T11:11:11Z"}'
            fields   = 'HCA;68036198;ERROR_FLAGS_4000 HEATER_TEMPERATURE_OUT_OF_RANGE TAMPERING;0;18.66;1111-11-11 11:11.11'
```

File: drivers/src/hydroclimav2.xmq [hunk 6]
```diff
@@ -253,8 +324,8 @@ driver {
...
-            fields   = 'HC3;93001021;ERROR_FLAGS_100 UNKNOWN_40;0;20.45;1111-11-11 11:11.11'
+            json     = '{"_":"telegram","media":"heat cost allocation","meter":"hydroclimav2","name":"HC3","id":"93001021","ambient_temperature_at_set_date_1_c":19.88,"ambient_temperature_at_set_date_2_c":20.86,"ambient_temperature_at_set_date_3_c":null,"average_ambient_temperature_c":20.45,"average_ambient_temperature_last_month_c":18.94,"average_heater_temperature_c":20.87,"average_heater_temperature_last_month_c":19.4,"consumption_at_set_date_1_hca":0,"consumption_at_set_date_2_hca":0,"consumption_at_set_date_3_hca":0,"consumption_at_set_date_4_hca":0,"consumption_hca":0,"max_ambient_temperature_c":20.75,"target_date":"2025-05-31","target_hca":0,"total_consumption_hca":34,"status":"EVENT_GROUP_B OPEN_CASE","timestamp":"1111-11-11T11:11:11Z"}'
+            fields   = 'HC3;93001021;EVENT_GROUP_B OPEN_CASE;0;20.45;1111-11-11 11:11.11'
```
Comment (Hymbajs) [inline drivers/src/hydroclimav2.xmq]: Same rename here.

```suggestion
            json     = '{"_":"telegram","media":"heat cost allocation","meter":"hydroclimav2","name":"HC3","id":"93001021","ambient_temperature_at_set_date_1_c":19.88,"ambient_temperature_at_set_date_2_c":20.86,"ambient_temperature_at_set_date_3_c":null,"average_ambient_temperature_c":20.45,"average_ambient_temperature_last_month_c":18.94,"average_heater_temperature_c":20.87,"average_heater_temperature_last_month_c":19.4,"consumption_at_set_date_1_hca":0,"consumption_at_set_date_2_hca":0,"consumption_at_set_date_3_hca":0,"consumption_at_set_date_4_hca":0,"consumption_hca":0,"max_ambient_temperature_c":20.75,"target_date":"2025-05-31","target_hca":0,"total_consumption_hca":34,"status":"EVENT_GROUP_B TAMPERING","timestamp":"1111-11-11T11:11:11Z"}'
            fields   = 'HC3;93001021;EVENT_GROUP_B TAMPERING;0;20.45;1111-11-11 11:11.11'
```

**Comment by strnadmichal** [comment]
Thank you @Hymbajs for collaboration. :) 
PR is ready to be merged @weetmuts . Thank you.

**Comment by weetmuts** [comment]
@strnadmichal Thanks! Merged.

https://wmbusmeters.org/analyze/2e44b0099861036853087a000020002F2F036E0000000F100043106A7D2C4A078F12202CB1242A06D3062100210000

**Comment by weetmuts** [comment]
@Hymbajs Can you drop me an email on oehrstroem@gmail.com ?
